manifests/key_pair.pp

```diff
@@ -0,0 +1,31 @@
+define certs::key_pair (
+  $source_dir,
+  $key_file,
+  $cert_file,
+  $key_ensure  = 'present',
+  $key_owner   = undef,
+  $key_group   = undef,
+  $key_mode    = undef,
+  $cert_ensure = 'present',
+  $cert_owner  = undef,
+  $cert_group  = undef,
+  $cert_mode   = undef,
+) {
+
+  file { $key_file:
```

Comment:
Should this always have `show_diff => false`?

Reply:
Good call out, there is a request for this out there in bug land.